apis/stacks/v1alpha1/types.go

```diff
@@ -305,6 +305,14 @@ type StackStatus struct {
 	ControllerRef                     *corev1.ObjectReference `json:"controllerRef,omitempty"`
 }
 
+// PackageMetadataSpec defines metadata about the stack application
+// and package contents
+type PackageMetadataSpec struct {
```

Comment:
this is a new type that isn't specifically included in the StackSpec, since it doesn't have meaning outside of the context of a package's app.yaml.  That makes sense to not include this data in a Stack CR (which has it's own CRD versioning process, independent of app.yaml schema/format/apiVersion), although that was counterintuitive to me for a second.